libs/base/Data/List.idr

```diff
@@ -954,3 +954,10 @@ export
 lengthMap : (xs : List a) -> length (map f xs) = length xs
 lengthMap [] = Refl
 lengthMap (x :: xs) = cong S (lengthMap xs)
+
+||| Proof that replicate produces a list of the requested length.
+export
+replicateLength : (n : Nat) -> length (replicate n x) = n
```

Comment:
I think we should stick to the convention introduced above: `lengthMap`  for the `length (map ...)`  lemma
so `lengthReplicate`  for the `length (replicate ...)` lemma.